src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -89,18 +88,16 @@ protected AssemblyLoadContext(bool isCollectible) : this(false, isCollectible)
         }
 
         internal AssemblyLoadContext(bool fRepresentsTPALoadContext, bool isCollectible)
-#else
-        protected AssemblyLoadContext() : this(false)
         {
-        }
-
-        internal AssemblyLoadContext(bool fRepresentsTPALoadContext)
+#if !FEATURE_COLLECTIBLE_ALC
+            if (isCollectible)
+            {
+                throw new NotSupportedException(Environment.GetResourceString("AssemblyLoadContext_Constructor_CollectibleNotSupported"));
```

Comment:
Think `InvalidOperationException` would be better, and for consistency with the exception thrown by `Unload`